lib/publiccloud/provider.pm

```diff
@@ -418,7 +418,7 @@ sub terraform_apply {
         my $sle_version = get_var('FORCED_DEPLOY_REPO_VERSION') ? get_var('FORCED_DEPLOY_REPO_VERSION') : get_var('VERSION');
         $sle_version =~ s/-/_/g;
         my $ha_sap_repo = get_var('HA_SAP_REPO') ? get_var('HA_SAP_REPO') . '/SLE_' . $sle_version : '';
-        my $suffix = sprintf("%04x", rand(0xffff));
+        my $suffix = sprintf("%04x", get_current_job_id());
```

Comment:
Carefull here. `"04x"` will turn into hexadecimal whatever it gets, and pad with zeroes at the left if the result is shorter than 4 characters:
```
# perl
$id = '1';
print sprintf("%04x", $id), "\n";
0001
# perl
$id = '16';
print sprintf("%04x", $id), "\n";
0010
```
Probably best to simply drop the `sprintf` and only use the job id.